test/functional/test-shell.md

```diff
@@ -0,0 +1,187 @@
+Test Shell for Interactive Environments
+=========================================
+
+This document describes the usage of the `TestShell` submodule in the functional
+test suite.
+
+The TestShell submodule extends the `BitcoinTestFramework` functionality to
+external interactive environments for prototyping and educational purposes. Just
+like `BitcoinTestFramework`, the TestShell allows the user to:
+
+* Manage regtest bitcoind subprocesses.
+* Access RPC interfaces of the underlying bitcoind instances.
+* Log events to the functional test logging utility.
+
+The `TestShell` can be useful in interactive environments where it is necessary
```

Comment:
nit: you write TestShell without backticks above and below; ideally settle on one style. Idem for BitcoinTestFramework.

Reply:
Fixed.